tests/reconfiguration.py

```diff
@@ -610,6 +611,143 @@ def run_join_old_snapshot(args):
                 pass
 
 
+def get_current_config(network):
+    tables, _ = network.get_latest_ledger_public_state()
+    nwcfgs_tn = "public:ccf.gov.nodes.network.configurations"
+    if nwcfgs_tn not in tables:
+        return None
+    nwcfgs = tables[nwcfgs_tn]
+    if ccf.ledger.WELL_KNOWN_SINGLETON_TABLE_KEY not in nwcfgs:
+        return None
+    current_ptr = json.loads(nwcfgs[ccf.ledger.WELL_KNOWN_SINGLETON_TABLE_KEY])
+    rid = current_ptr["rid"].to_bytes(8, "little")
+    if rid not in nwcfgs:
+        return None
+    current_cfg = json.loads(nwcfgs[rid])
+    return current_cfg
+
+
+def get_current_nodes_table(network):
+    tables, _ = network.get_latest_ledger_public_state()
+    tn = "public:ccf.gov.nodes.info"
+    r = {}
+    for nid, info in tables[tn].items():
+        r[nid.decode()] = json.loads(info)
+    return r
+
+
+def get_joining_node_dates(network, tx_id_before, tx_id_after, node_id):
+    pending_at = 0
+    learner_at = 0
+    trusted_at = 0
+
+    for i in range(tx_id_before.seqno, tx_id_after.seqno + 1):
+        tables = network.get_ledger_public_state_at(i)
+        ntn = "public:ccf.gov.nodes.info"
+        if ntn in tables:
+            for k, v in tables[ntn].items():
+                if k.decode() == node_id:
+                    jv = json.loads(v)
+                    if "status" in jv:
+                        if jv["status"] == "Pending":
+                            pending_at = i
+                        elif jv["status"] == "Learner":
+                            learner_at = i
+                        elif jv["status"] == "Trusted":
+                            trusted_at = i
+
+    return pending_at, learner_at, trusted_at
+
+
+def run_migration_tests(args):
+    if args.reconfiguration_type != "OneTransaction":
+        return
+
+    txs = app.LoggingTxs("user0")
```

Comment:
`txs` is unused?

Reply:
No it isn't: `txs=txs,` in the call to `network(...)`